src/ui/TableHeadCell.tsx

```diff
@@ -0,0 +1,47 @@
+import { TableCell as MuiTableCell, TableCellProps, TableSortLabel } from "@mui/material";
+import { useIntl } from "react-intl";
+
+type Props = {
+  columnId: string;
+  label: string;
+  sort?: boolean;
+  order?: "asc" | "desc";
+  orderBy?: string;
+  onRequestSort?: (event: React.MouseEvent<unknown>, property: string) => void;
+} & Pick<TableCellProps, "sx">;
+
+export const TableHeadCell = ({
+  columnId,
+  label,
+  sort = true,
+  order,
+  orderBy,
+  onRequestSort,
+  sx,
+}: Props) => {
+  const intl = useIntl();
+
+  const createSortHandler = (property: string) => (event: React.MouseEvent<unknown>) => {
+    if (!onRequestSort) {
+      return;
+    }
+    onRequestSort(event, property);
```

Comment:
```suggestion
   onRequestSort?.(event, property);
```
optional chaining to ensure that the function is not null or undefined